opencensus/metrics/transport.py

```diff
@@ -61,19 +62,23 @@ def func(*aa, **kw):
             except Exception:
                 logger.exception("Error handling metric export")
 
-        super(MetricExporterTask, self).__init__(interval, func, args, kwargs)
+        super(PeriodicMetricTask, self).__init__(interval, func, args, kwargs)
 
 
-def get_exporter_thread(metric_producer, exporter, interval=None):
+def get_exporter_thread(metric_producers, exporter, interval=None):
```

Comment:
I have a design question, do we intend to have one `MetricProducer` taking multiple metrics, or multiple `MetricProducer` (which I guess is the case)? If it is the latter, what's the main consideration of having multiple producers (e.g. just to categorize by ownership, or ext lib, or we expect each producer to export individually?)

Reply:
Yes we intend to take multiple MetricProducer. The idea is simply as you noted to categorize ownership of the types of metrics that each Producer produces. My thought was to separate these based on logic, so if a developer would like to produce some other set of metrics using a different architecture (stats, gauges, etc.), they can simply create their own MetricProducer and implement "get_metrics".

Reply:
Got it, thanks.

Reply:
@c24t would you take a look at the changes to the core library? Thanks.

Reply:
This change looks fine, but the other option is to add a metric producer class that calls other metric producers under the hood. Since we're likely to make a similar change for exporters we might want to revisit this later. But note that this change changes the API, the other way doesn't.